### PR TITLE
Add character endpoints

### PR DESCRIPTION
**Pull request by Buried-In-Code**
Some of the notable changes:
 - Added Character and CharacterList endpoints
 - Changing the docstrings formatting to Google
 - Added Attribute list to data classes
 - Renaming the project folder from `Simyan` to `simyan`
 - Moved similar modules into package directory
   - *e.g. Moved `issue.py` and `issue_list.py`to `issue/*.py`*
 - Added `create_session()`.
   - *Deprecated: `api()`*
 - Made `Session.get` protected
   - *e.g. `Session._get`*
 - Renamed `ApiError` to `APIError`
 - Renamed `CVType` to `ComicVineResource`
 - Renamed `SqliteCache` to `SQLiteCache`
 - Renamed `Issue.character_deaths` to `Issue.deaths`
 - Renamed `Creator.created_characters` to `Creator.characters`

### CODE REVIEW

File: simyan/__init__.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+"""simyan package entry file."""
+__version__ = "0.6.0"
+__all__ = ["__version__", "create_session", "api", "SQLiteCache"]
```
Comment (bpepple) [inline simyan/__init__.py:3]: Why are `__version__` & `SQLiteCache` being exposed?

**Review by bpepple — COMMENTED** [review]
Gave a quick glance to see what this pull request entailed, and a couple of quick thoughts:

1. In the future, to make reviewing easier I'd split the pull request into separate chunks based on what's being proposed. That way if one feature has problems that need to be fixed and/or rejected the whole pull request doesn't have to be reworked. For example, this one could have been split into:
 - Character Endpoints
 - Docs Changes
 - Exports Improvements
2. I'm not quite sure what benefit moving the various resources (issues, publisher, etc)  to separate folders provides. Seems to make it unnecessary difficult/cumbersome to navigate to project files. For example, let's say you wanted to added functionality to write/read metadata files from a comic file it would make sense to move *that* to a separate module, but in this case, there are very few files and they all deal with Comic Vine data handling. Anyway, just my 2¢.
3. The import changes you've made for the most part aren't really necessary. For example, there are *only* public names in the `Character` files, so declaring `__all__` is completely unnecessary.  The only files that would possible need `__all__` defined would be the project `__init__py` and `session.py`. Usually, when a project is under early development and still tweaking the API it makes more sense to use the `_` prefix convention in lieu of `__all__`, but if you do choose to use `__all__` it doesn't make sense to *also* use the `_` prefix.

Anyway, that's just my initial thoughts, I'll give a more thorough review when I get a block of free time this weekend.

**Comment by bpepple** [comment]
>     3. The inclusion of `__all__` in each of the packages fulfill the same purpose as including it for the `SQLiteCache` in `__init__.py` just making the imports simpler. But the whole public/private part of python is new to me so if I'm doing it wrong, happy to adjust

Right, I think you might be misunderstanding what `__all__` is used for. It's for publicly exposing names, not to make imports more simple. So, for example users of _simyan_ would expect to have `api` or `create_session` in the main project `__init__.py` to be exposed, but *not* `SQLiteCache` since that is an internally used module. Almost all the other modules have only public name, so defining `__all__` in them is sorta pointless.

**Comment by Buried-In-Code** [comment]
So after removing each `__all__` locally flake8 doesn't like the imports being there and "F401: imported but unused" (they are used by session).  
Despite there only being public names in the packages making declaring __all__ is completely unnecessary, having it solves the flake8 issue.

**Pull request merged**